extensions/amp-list/0.1/amp-list.js

```diff
@@ -672,41 +691,77 @@ export class AmpList extends AMP.BaseElement {
    * @private
    */
   setLoadMore_() {
-    if (!this.loadMoreSrc_ && !this.loadMoreOverflow_) {
+    if (!this.loadMoreSrc_ && !this.loadMoreButton_) {
       return;
     }
     const triggerOnScroll = this.element.getAttribute('load-more') === 'auto';
     if (triggerOnScroll) {
       this.maybeSetupLoadMoreAuto_();
     }
-    if (this.loadMoreOverflow_) {
+    if (this.loadMoreButton_) {
       this.mutateElement(() => {
-        this.loadMoreOverflow_.classList.toggle('amp-visible', true);
-        listen(this.loadMoreOverflow_, 'click', () => this.loadMoreCallback_());
+        this.loadMoreButton_.classList.toggle('amp-visible', true);
+        this.unlistenLoadMoreButton_ = listen(this.loadMoreButton_, 'click',
+            () => this.loadMoreCallback_());
       });
     }
-    if (!this.loadMoreOverflow_ && !triggerOnScroll) {
+    if (!this.loadMoreButton_ && !triggerOnScroll) {
       user().error(TAG,
           'load-more is specified but no means of paging (overflow or ' +
           'load-more=auto) is available', this);
     }
   }
 
   /**
+   * Called when a fetch fails under load-more. Shows the load-more-button
+   * element and triggers a reloading of the failed src on click.
+   * @private
+   */
+  setLoadMoreReload_() {
+    if (this.loadMoreButton_) {
+      this.mutateElement(() => {
+        this.loadMoreButton_.classList.toggle('amp-visible', true);
+        this.unlistenLoadMoreButton_ = listen(this.loadMoreButton_, 'click',
+            () => this.loadMoreReloadCallback_());
```

Comment:
can `loadMoreReloadCallback` and `loadMoreCallback_` become the same thing? they are almost doing the same functionality. (at minimum the common part should be refactored)

Reply:
Done